src/libraries/Microsoft.Extensions.Logging.EventSource/src/EventSourceLogger.cs

```diff
@@ -104,14 +104,16 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     exceptionJson = ToJson(exceptionInfoData);
                 }
                 IReadOnlyList<KeyValuePair<string, string>> arguments = GetProperties(state);
+                string message = formatter(state, exception);
```

Comment:
>formatter [](start = 33, length = 9)

I am seeing this is done in line 64 too. Is it possible we can have a case having _eventSource enabled with FormattedMessage and JsonMessage in same time? 

I am asking this to see if we can optimize calling `string message = formatter(state, exception);` once